.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,222 @@
+
+1.  The last line in each chapter includes the file `links.md`, which gives symbolic names to all the external links used in the book.
+    Please always use `[text][label]` to refer to links and put the URL itself in this file to ensure consistency between chapters.
+
+1.  Please use the free desktop version of [draw.io](https://www.draw.io/) to draw diagrams.
+    Save the source as `figures/stem/stem.drawio` and export drawings as all of PDF, PNG, and SVG.
+
+1.  Since LaTeX doesn't understand SVG images, hand-drawn diagrams are included as follows:
+    -   Create an R code block with the header `{r stem-label, echo=FALSE, fig.cap="Some Caption"}`, which means:
+        -   Give this figure the ID `fig:stem-label`.
+            (Bookdown automatically puts `fig:` in front of figure labels,
+            though it doesn't prefix section labels with anything.)
+        -   Don't show the R code used to load the image, just its output (i.e., the image).
+        -   Give the figure the specified caption.
+    -   Use `insert_graphic("figures/stem/filename.ext")` to include your image. If you are using a `.pdf` image, make sure that you have the same file but as a `.svg` as well. 
+
+1.  Use `@Name1234` to refer to bibliography entries.
+
+1.  Use `\@ref(label)` to refer to labels for sections and figures.
+    Note that:
+    -   This only inserts the section number, not the words `Chapter`, `Appendix`, `Section`, or `Figure`.
+    -   Those are round parentheses, not curly braces.
+    -   If the figure's chunk ID is `stem-label`, use `fig:stem-label` to refer to it.
+
+1.  Glossary entries are in `gloss.md`, which is plain Markdown rather than R Markdown.
+    -   To refer to a glossary entry, use a direct link of the form `[text]``(glossary.html#term-label)`.
+    -   Glossary definitions are set in bold and use an HTML anchor tag to provide the ID.
+        We should find a more elegant way to do this.
+
+1.  Many code fragments are accompanied by an HTML comment that contains `src="stem/filename.ext"`.
+    This identifies the path to that file within the `./src` folder.
+    We should find a more elegant way to handle these references.
+
+1.  There are also some HTML comments containing the word `noindent` left over from formatting with an earlier template.
+    These were used to prevent indentation of the first line of continuation paragraphs.
+    We will find a more elegant way to handle this as we get closer to production.
+    
+    -   Do *not* use `View` in your R snippets,
+        as Travis will fail when it tries to launch a viewer from the command line.
+    
+
+## Using Python Code Chunks
+
+R users should be already familiar with using R Markdown. The process for using
+Python in R Markdown is familiar straight forward. When you want to include some
+code and its output, use code chunks like:
+    
+    ```{python}
+    python code ...
```

Comment:
```suggestion
    # Python chunks will also be syntax highlighted
    for num in range(5):
        print(num)
```

Reply:
I guess this maybe is unnecessary... i was thinking that this section was in the book so it would be nice to show higlighting, but it is just in the contributing which won't recognize `{lang}` anyways. Oh well